### DISCUSSION TITLE
MSI Install under UAC with 1603 error still installs

### DISCUSSION THREAD

**Discussion by hegsie** [wixtoolset/issues#7961] [Questions] [answered]
Hi,

I have an MSI written with Wix4 and when I try and install it from the C:\Users\<username> folder, with admin elevation it prompts for UAC correctly but then fails inside a CA with 1603, the odd thing is however the MSI is registered in the apps & features as installed even with the failure in the logs.

`Wix4JsonFile:  Updating the json $.Provider.HostName with values backbonedv.
MSI (s) (84!48) [09:40:08:462]: Closing MSIHANDLE (172) of type 790531 for thread 30536
MSI (s) (84!48) [09:40:08:463]: Creating MSIHANDLE (173) of type 790531 for thread 30536
Wix4JsonFile:  GetLastError returned 5, Access is denied.
 @ creating the output stream
MSI (s) (84!48) [09:40:08:463]: Closing MSIHANDLE (173) of type 790531 for thread 30536
MSI (s) (84!48) [09:40:08:463]: Creating MSIHANDLE (174) of type 790531 for thread 30536
Wix4JsonFile:  HRESULT hr -2147024891
MSI (s) (84!48) [09:40:08:464]: Closing MSIHANDLE (174) of type 790531 for thread 30536
MSI (s) (84!48) [09:40:08:464]: Creating MSIHANDLE (175) of type 790531 for thread 30536
Wix4JsonFile:  Error 0x80070005: Failed while navigating path: C:\Program Files\SEFE\Trading.Coral.InvoiceCollector\appSettings.json for row: SubscriberHostName
MSI (s) (84!48) [09:40:08:464]: Closing MSIHANDLE (175) of type 790531 for thread 30536
MSI (s) (84!48) [09:40:08:464]: Closing MSIHANDLE (153) of type 790540 for thread 30536
MSI (s) (84!48) [09:40:08:464]: Creating MSIHANDLE (176) of type 790531 for thread 30536
Wix4JsonFile:  Closed PMSIHANDLE hView
MSI (s) (84!48) [09:40:08:465]: Closing MSIHANDLE (176) of type 790531 for thread 30536
MSI (s) (84!48) [09:40:08:465]: Closing MSIHANDLE (154) of type 790531 for thread 30536
MSI (s) (84!48) [09:40:08:465]: Creating MSIHANDLE (177) of type 790531 for thread 30536
Wix4JsonFile:  Closed PMSIHANDLE hRec
MSI (s) (84!48) [09:40:08:465]: Closing MSIHANDLE (177) of type 790531 for thread 30536
MSI (s) (84!48) [09:40:08:465]: Closing MSIHANDLE (147) of type 790541 for thread 30536
CustomAction Wix4JsonFile_X64 returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
MSI (s) (84:68) [09:40:08:466]: Closing MSIHANDLE (146) of type 790542 for thread 29848
Action ended 09:40:08: Wix4JsonFile_X64. Return value 3.
Action ended 09:40:08: INSTALL. Return value 3.
MSI (s) (84:98) [09:40:08:471]: Closing MSIHANDLE (145) of type 790542 for thread 29848
MSI (s) (84:98) [09:40:08:476]: MainEngineThread is returning 1603
MSI (s) (84:C0) [09:40:08:483]: RESTART MANAGER: Session closed.
MSI (s) (84:C0) [09:40:08:483]: No System Restore sequence number for this installation.
MSI (s) (84:C0) [09:40:08:487]: User policy value 'DisableRollback' is 0
MSI (s) (84:C0) [09:40:08:487]: Machine policy value 'DisableRollback' is 0
MSI (s) (84:C0) [09:40:08:487]: Incrementing counter to disable shutdown. Counter after increment: 0
MSI (s) (84:C0) [09:40:08:487]: Note: 1: 1402 2: HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Installer\Rollback\Scripts 3: 2 
MSI (s) (84:C0) [09:40:08:488]: Note: 1: 1402 2: HKEY_LOCAL_MACHINE\Software\Microsoft\Windows\CurrentVersion\Installer\Rollback\Scripts 3: 2 
MSI (s) (84:C0) [09:40:08:489]: Decrementing counter to disable shutdown. If counter >= 0, shutdown will be denied.  Counter after decrement: -1
MSI (s) (84:C0) [09:40:08:489]: Destroying RemoteAPI object.
MSI (s) (84:0C) [09:40:08:489]: Custom Action Manager thread ending.
MSI (c) (1C:80) [09:40:08:491]: Back from server. Return value: 1603
MSI (c) (1C:80) [09:40:08:491]: Decrementing counter to disable shutdown. If counter >= 0, shutdown will be denied.  Counter after decrement: -1
MSI (c) (1C:80) [09:40:08:491]: PROPERTY CHANGE: Deleting SECONDSEQUENCE property. Its current value is '1'.
Action ended 09:40:08: ExecuteAction. Return value 3.
Action ended 09:40:08: INSTALL. Return value 3.
=== Logging stopped: 18/01/2024  09:40:08 ===
MSI (c) (1C:80) [09:40:08:493]: Note: 1: 1708 
MSI (c) (1C:80) [09:40:08:493]: Note: 1: 2205 2:  3: Error 
MSI (c) (1C:80) [09:40:08:493]: Note: 1: 2228 2:  3: Error 4: SELECT `Message` FROM `Error` WHERE `Error` = 1708 
MSI (c) (1C:80) [09:40:08:493]: Note: 1: 2205 2:  3: Error 
MSI (c) (1C:80) [09:40:08:493]: Note: 1: 2228 2:  3: Error 4: SELECT `Message` FROM `Error` WHERE `Error` = 1709 
MSI (c) (1C:80) [09:40:08:493]: Product: Trading.Coral.InvoiceCollector -- Installation failed.

MSI (c) (1C:80) [09:40:08:494]: Windows Installer installed the product. Product Name: Trading.Coral.InvoiceCollector. Product Version: 1.0.0.0. Product Language: 1033. Manufacturer: SEFE Group. Installation success or error status: 1603.

MSI (c) (1C:80) [09:40:08:494]: Grabbed execution mutex.
MSI (c) (1C:80) [09:40:08:494]: Cleaning up uninstalled install packages, if any exist
MSI (c) (1C:80) [09:40:08:496]: MainEngineThread is returning 1603
=== Verbose logging stopped: 18/01/2024  09:40:08 ===
`

Any ideas or thoughts would be appreciated

Thanks

**Comment by robmen** [comment, marked as answer]
Your custom action failed. You don't provide enough information about the custom action, how it is scheduled to know much more. My spider-sense guesses it isn't deferred and ignores its return code.

**Reply by hegsie** [reply]
Thankyou, this fixed the issue, much appreciated
